examples/analytics/emr-on-eks/README.md

```diff
@@ -22,45 +22,53 @@ _Note: Currently Amazon Prometheus supported only in selected regions. Please se
 
 Clone the repository
 
-```
+```sh
 git clone https://github.com/aws-ia/terraform-aws-eks-blueprints.git
 ```
 
 Navigate into one of the example directories and run `terraform init`
 
-```
+```sh
 cd examples/analytics/emr-on-eks
 terraform init
 ```
 
 Set AWS_REGION and Run Terraform plan to verify the resources created by this execution.
 
-```
+```sh
 export AWS_REGION="<enter-your-region>"
 terraform plan
 ```
 
-Deploy the pattern
+**Deploy the pattern**
+
+Deploy VPC, EKS cluster with Node groups and Kubernetes Add-ons with `--target` option
 
+```sh
+terraform apply -target="module.aws_vpc"
```

Comment:
I believe for apply we want to advise a standard `terraform apply` otherwise you can get some strange results and transient failures. On the destroy side though, we definitely want to destroy addons first

Reply:
Terraform apply failed in the past when we don't have `depends_on`  on the node groups. @Zvikan can add some insights here. I know it works for some examples but i am not sure if it works for all examples

Now, I have removed that `depends_on` line to avoid a major issue of replacing the managed policies. Hence i added terraform apply also with `--target`. If our E2E parallel tests confirms that it woks with one APPLY then we can change that.

I don't really like this `--target` option and we have no other option for examples as the users expect the whole example to work with one single TF APPLY. I wouldn't recommend this option for production. 

Customers/users needs to split this examples and create multiple state files for VPC, EKS & NodeGroups and Kubernetes Add-ons before they actually use in Prod.

Reply:
When testing with e2e yesterday using the split up targets for applies, I would frequently see transient failures like `Error: error reading Route Table (rtb-07e46cd0c3469e326): couldn't find resource`. But using a standard `terraform apply`, the results were as expected.

However, the targeted destroy is still needed - especially for those addons that create additional AWS networking resources

Here is the current [e2e setup](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/.github/workflows/e2e-parallel-full.yml#L63-L77) that is working as [expected](https://github.com/aws-ia/terraform-aws-eks-blueprints/actions/runs/2393294684) for reference

Reply:
Thanks 👍 I will update the PR with one apply

Reply:
Previously we were running apply with `-target` with success, I do recall seeing error like `Error: error reading Route Table (rtb-07e46cd0c3469e326): couldn't find resource` that you mentioned, but they were yet another intermittent issue.
I still think it's worth to use `--target` for apply as-well, it won't be impactful as for destroy, but it may still cause some delays (which may lead to timeout exceptions as-well) during creation.


Scenario example that I can think of is if VPC resource (e.g. NAT gateway) is taking its time to be created (for whatever reason), other modules may start create their own resources (e.g. self-managed-node) which may fail due to timeout of creation which is mainly caused by NAT, as the nodes were not able to connect to the cluster for example.

Will take a look at the failures around the "route table can't find resource issue" if it's reproducible more easily today.